tjpcov_new/covariance_calculator.py

```diff
@@ -0,0 +1,54 @@
+from .covariance_io import CovarianceIO
+from . import covariance_from_name
+import numpy as np
+
+
+class CovarianceCalculator(CovarianceIO):
```

Comment:
This really is all great, just a question - would it make sense to keep the CovarianceIO and the CovarianceCalculators actually as separate classes?  The CovarianceIO / Builder can just instantiate and call the calculators within the builder class.  That way you could reduce some of the inheritance?  If this doesn't make sense then ignore!  Looks great!

Reply:
Thanks! `CovarianceIO` is now a separate class.